packages/sign-client/src/controllers/engine.ts

```diff
@@ -2941,4 +2974,56 @@ export class Engine extends IEngine {
       }
     }
   };
+
+  private shouldSetTVF = (
+    protocolMethod: JsonRpcTypes.WcMethod,
+    params: JsonRpcTypes.RequestParams["wc_sessionRequest"],
+  ) => {
+    if (!params) return false;
+    if (protocolMethod !== "wc_sessionRequest") return false;
+    const { request, chainId } = params;
+    const namespace = parseChainId(chainId).namespace;
+    if (!(namespace in TVF_METHODS)) return false;
+    const tvfMethods = TVF_METHODS[namespace as keyof typeof TVF_METHODS];
+    if (!tvfMethods) return false;
+    return tvfMethods.includes(request.method);
```

Comment:
Do you think it'd make sense to introduce a validation library with clear schemas for handling validation ?

Reply:
the parsing is simplified now so probably overkill to export it into its own library. if we expand the handling at later point, why not